hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java

```diff
@@ -720,6 +670,101 @@ public boolean delete(Path f, boolean recursive) throws IOException {
     return result;
   }
 
+  private boolean deleteBucket(Path f, boolean recursive, OFSPath ofsPath)
+      throws IOException {
+    // check status of normal bucket
+    try {
+      getFileStatus(f);
+    } catch (FileNotFoundException ex) {
+      // remove orphan link bucket directly
+      if (isLinkBucket(f, ofsPath)) {
```

Comment:
Within this catch block we can just check `if isLinkBucket == false` and return `LOG.warn(Path does not exist)` with `false`.  Outside the catch block the bucket is checked if link, if true, then the orphan bucket is removed from volume and returned.

Reply:
@neils-dev 
isLinkBucket performs remote call to OM. So checking first as false in exception, and again checking outside for link bucket bucket will have 2 remote call. So I think this is not required to be refactored for performance.

Reply:
Thanks for pointing out the remote calls.  Makes sense.